beamer/state_machine.py

```diff
@@ -235,7 +265,7 @@ def _handle_claim_made(event: ClaimMade, context: Context) -> HandlerResult:
         return False, None
 
     events: list[Event] = []
-    if not claim.is_invalidated_l1_resolved and claim.invalidation_tx is not None:
+    if _invalidation_ready_for_l1_relay(claim, request.target_chain_id, context.request_manager):
```

Comment:
If the condition is false, then `InitiateL1InvalidationEvent` will not be created. And if no other ClaimMade event will ever arrive, then it will not trigger L1 resolution. 

I think what would happen now is, that as long as the above condition is false, the agent will counter challenge again, Probably leading to drain all the funds (depending on the attacker).

Reply:
Well, I'm not exactly sure about the solution. But currently what happens is: 

As long as the condition is not true, which is finalization time has not passed yet, then we will ultimately engage in the challenge again. That's not what we want. As soon as the agent has gathered enough funds (l1_criteria_fulfilled), we want him to wait and not counter challenge. 

That's what happened before by moving to the l1_resolved state. But since we are not emitting the event before finalization time is passed we will either stay in `claimer_winning` or `challenger_winning` and this will lead to another challenge transaction.

Reply:
Nice find, I pulled out the wait into the handling of said events.